src/pages/debrief/cat-b/debrief.cat-b.selector.ts

```diff
@@ -0,0 +1,214 @@
+import { forOwn, transform, endsWith } from 'lodash';
+import { CatBUniqueTypes } from '@dvsa/mes-test-schema/categories/B';
+import {
+  manoeuvreTypeLabels,
+  manoeuvreCompetencyLabels,
+} from '../../../shared/constants/competencies/catb-manoeuvres';
+import { ManoeuvreTypes } from '../../../modules/tests/test-data/test-data.constants';
+import { CompetencyOutcome } from '../../../shared/models/competency-outcome';
+import {
+  MultiFaultAssignableCompetency,
+  CommentedCompetency,
+  CommentSource,
+} from '../../../shared/models/fault-marking.model';
+import { CompetencyDisplayName } from '../../../shared/models/competency-display-name';
+import { getSeriousOrDangerousFaults } from '../debrief.selector';
+// TODO - A lot of these are used in multiple places (Debrief, View Test Result, Office),
+// should be refactored into a more common area.
+export const getManoeuvreFaults = (
```

Comment:
How did we end up with this implementation?!
The Imitation Game could have been about cracking this function, not the Enigma machine..

Reply:
I blame the data structure for holding control and observation faults. Should never have been separate properties for control and observation. However, this method was not introduced by this PR. It can be added to the tech debt confluence page though.

Reply:
Decided to do a minor refactor to give a bit more clarity to what this method does.

Reply:
I like how you convinced yourself to do it in the course of 3 hours 😄